core/src/main/java/org/apache/iceberg/rest/RESTMetricsReporter.java

```diff
@@ -51,15 +53,17 @@ public void report(MetricsReport report) {
       return;
     }
 
-    try {
-      client.post(
-          metricsEndpoint,
-          ReportMetricsRequest.of(report),
-          null,
-          headers,
-          ErrorHandlers.defaultErrorHandler());
-    } catch (Exception e) {
-      LOG.warn("Failed to report metrics to REST endpoint {}", metricsEndpoint, e);
-    }
+    Tasks.range(1)
+        .executeWith(ThreadPools.getWorkerPool())
```

Comment:
I don't think we should be using the worker pool here since that's used for planning operations. I wouldn't want to possibly interfere with that with metrics reporting which is basically best effort. Should we just define  an exiting thread pool `metrics-publisher-pool` or something like that and then just use that.